qa/tasks/cephfs/test_multifs_auth.py

```diff
@@ -155,37 +171,37 @@ def tearDown(self):
 
         super(type(self), self).tearDown()
 
-    def _create_client(self, perm, fsname=False, cephfs_mntpt='/'):
+    def _gen_caps(self, perm, both_fsnames=False, cephfs_mntpt='/'):
         moncap = 'allow r'
         osdcap = gen_osd_cap_str(((perm, self.fs1.name),
                                   (perm, self.fs2.name)))
-        if fsname:
+
+        if both_fsnames:
             mdscap = gen_mds_cap_str(((perm, self.fs1.name, cephfs_mntpt),
                                       (perm, self.fs2.name, cephfs_mntpt)))
         else:
             mdscap = gen_mds_cap_str(((perm, None, cephfs_mntpt),
                                       (perm, None, cephfs_mntpt)))
 
-        keyring = self.create_client(self.client_id, moncap, osdcap, mdscap)
-        keyring_paths = []
-        for mount_x in self.mounts:
```

Comment:
This is changed to

> keyring_path = self.mount_a.client_remote.mktemp(data=keyring)

in remount_with_new_client(). Does this work with a test that uses more than one client since the above _always_ uses `self.mount_a`?

Reply:
> This is changed to
> 
> > keyring_path = self.mount_a.client_remote.mktemp(data=keyring)
> 
> in remount_with_new_client().

You mean this line - https://github.com/ceph/ceph/pull/50665/files/06a42bd4ea7e46676e3d17f22b3fb21f1b05ef9c#diff-8d323d162bb162ad0244529eef8eabce74a318645c3ebe81f3b92f3abdde25f2R194?

> Does this work with a test that uses more than one client since the above _always_ uses `self.mount_a`?

Yes because we have separate line to write keyring for mount_b. `        keyring_path = self.mount_b.client_remote.mktemp(data=keyring)`

Reply:
ACK